wiki/Upgrading-Target-Sdk-Guide.md

```diff
@@ -0,0 +1,148 @@
+## Overview
+
+Updating Oppia Android's [target SDK](https://developer.android.com/guide/topics/manifest/uses-sdk-element#target) provides an explicit signal to Android OS versions at or above the new target SDK level that the app should work correctly for that platform.
+
+The target SDK version, unlike the compile SDK version, is specifically a runtime behavior signal. That means it enables functionality only observable by opening and running the app on the corresponding, or newer, versions of the OS. When Android makes changes that could break compatibility with older apps, they will usually gate this behind the target SDK level so that apps can have time to upgrade without users being unable to use them after they themselves upgrade to the newer Android version. Not every version of Android introduces these compatibility breakages, and not every potential breakage will affect Oppia Android.
+
+This guide describes the high-level process for upgrading the app to a newer version of Android, and how to do it in a way that should reduce the risk of introducing breakages to users.
+
+## Upgrade Process
+
+```mermaid
+flowchart TD
+    A(Part 1: Identifying the need to upgrade) -->|Tracking issue exists| B
+    B(Part 2: Auditing the Android OS changelog) -->|Audit complete| C
+    C(Part 3: Testing the app & filing problems) -->|Testing finished & issues filed| D
+    D(Part 4: Fixing and stabilizing support) -->|All issues fixed| E
+    E(Part 5: Submitting the upgrade) -->|Compile/target SDK PR submitted| F
```

Comment:
What exactly is being cut off for you?

Here's what I see:

![image](https://github.com/user-attachments/assets/cfc3c11a-f902-417b-8c4e-797aa159d1f4)

Reply:
I viewed the page using View File and it renders as below:

![image](https://github.com/user-attachments/assets/c07ebfd8-b8be-4d7e-bf58-86da73af8fc9)
 
Am I viewing it correctly?

Reply:
ah, maybe due to varied zoom?

https://github.com/user-attachments/assets/e74f2c87-c4d3-4ef6-b988-fe8f396cbf16

Reply:
This is actually really interesting. What browser and browser version are you using? This might be a browser-specific GitHub and/or Mermaid bug as zoom doesn't affect things at all for me. From inspecting the graph, it's a generated SVG being rendered.

I'm using Chrome 129.

Reply:
I initially tested it on Edge 128, but now I also tested it with
- Chrome 120
- Edge 128
- Chrome 129 (Updated)

seems like on my end, they were still reproducible.

https://github.com/user-attachments/assets/967ae445-7159-4abf-86b3-9039787f6964

Reply:
Could any of the OS specific rendering make these differences, I tested it by installing Mint on VM and resembles as you mentioned but the same with Windows 11 (Edge 128) renders it differently. Also, this issue - https://github.com/typora/typora-issues/issues/3291#issuecomment-680117090 seems to mention a similar issue for MacOs.

https://github.com/user-attachments/assets/c8d089e6-682d-4d24-b3fb-441be3a1b411

Reply:
I'm not 100% sure honestly. Browser rendering can get very complicated, so I can only guess on potential compatibility. I tried searching the Mermaid GitHub repo for similar issues and didn't find a perfect match (though there were other text cut-off issues for separate reasons that were addressed). The issue you linked is much closer to what we're observing here than anything I found.

I'm not opposed to adding breaks, but I have little confidence that will fix the problem correctly since a separate display situations might still result in it being cut off. Or, put another way, how much text can we actually fit on a line before it's likely to be cut off? This is further complicated by us not using a monospaced font so line width won't be exactly correlated to number of characters.

Do you have any suggestions on what we could do here? I'm wondering if a different diagram format or maybe just trying to shorten the text would be best (though I thought of that and couldn't come up with good alternatives at the time).

Reply:
I tried applying the `text-wrap` property to the node, and it appears to be working. Could this be a viable / consistent workaround -- PTAL? I'll also test it on other browsers to verify its behavior.


https://github.com/user-attachments/assets/6147dcd5-6824-4940-9747-aea832323d2c

>Tested on Edge 128

Reply:
Just mentioning that I have not experienced any cutoff issues on chrome or safari even when zooming.

Reply:
Very nice suggestion! Applied the suggested changes. PTAL @Rd4dev.

Reply:
Thanks, @BenHenning.